N1C WAF Security Dashboard Release (#1834)
nginx/documentation: content/nginx-one-console/waf-integration/waf-security-dashboard/security-monitoring-overview.md

---
nd-content-type: concept
nd-docs: DOCS-000
nd-product: NONECO
title: Security monitoring overview
description: "What the F5 WAF for NGINX security monitoring module is, the data pipeline behind it, and what you can do with the data."
weight: 100
toc: true
nd-keywords: "security monitoring, security dashboard, F5 WAF for NGINX, security events, analytics, NGINX One Console"
nd-summary: >
  Security monitoring in NGINX One Console centralizes security events from F5 WAF for NGINX instances and exposes them through dashboards and an analytics API.
  Use this document to understand what the dashboard shows, where the data comes from, and how it is scoped and retained.
  This document is conceptual; for setup steps see Set up security monitoring.
nd-audience: operator
---

> Security monitoring brings F5 WAF for NGINX events from every connected instance into a single place in NGINX One Console. Before you set it up, it helps to understand what data the system collects, how it gets there, and how it is organized.

Reviewer: ```suggestion
Security monitoring brings F5 WAF for NGINX events from every connected instance into a single place in NGINX One Console. Before you set up security monitoring, read this document to understand what data the system collects, how the data flows, and how the data is organized.
```

Reviewer 2: slightly different:

Security monitoring brings F5 WAF for NGINX events from every connected instance into a single place in NGINX One Console. This page explains what data the system collects, how it flows, and how it is organized.

## What is security monitoring?

> Security monitoring is the NGINX One Console module that ingests F5 WAF for NGINX security events from your data planes, stores them centrally, and exposes them through a security dashboard and an analytics API. It gives you a single view of attacks, violations, and triggered signatures across every NGINX Plus instance you have connected to NGINX One Console.
> 
> This document covers what the module is and how data flows through it. For deployment steps, see [Set up security monitoring]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/set-up-security-monitoring.md" >}}). For details on what each dashboard widget shows, see the [dashboard metrics reference]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/dashboard-metrics-reference.md" >}}).

Reviewer: ```suggestion
Security monitoring is the NGINX One Console module that ingests F5 WAF for NGINX security events from your data planes and stores them centrally. Security monitoring exposes ingested events through a security dashboard and an analytics API. Security monitoring gives you a single view of attacks, violations, and triggered signatures across every NGINX Plus instance connected to NGINX One Console.

This document covers what security monitoring is and how data flows through it. For deployment steps, see [Set up security monitoring]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/set-up-security-monitoring.md" >}}). For details on what each dashboard widget shows, see the [dashboard metrics reference]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/dashboard-metrics-reference.md" >}}).
```

Reviewer 2: I am not sure about this one. "It" replaced with "Security monitoring" seems like a lot. Keeping the original but let me know why.

Reviewer: Our guidance is to avoid vagueness. This change also help agentic LLMs understand two separate ideas

---

## How security events reach the dashboard

> 

Reviewer: would a diagram for this be helpful here?

Reviewer 2: May be. Don't have one handy that can be made public. :nit for me

The data pipeline has four stages:

1. **Detection.** F5 WAF for NGINX inspects requests on the data plane and produces a security log entry whenever a request matches a violation, signature, or threat campaign.
2. **Forwarding.** F5 WAF for NGINX writes the entry over syslog (port `1514` on localhost) using the [`secops_dashboard` log profile]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/default-log-profile.md" >}}). NGINX Agent's OpenTelemetry collector receives it through the `tcplog/nginx_app_protect` receiver.
3. **Transport.** The collector batches events and exports them to NGINX One Console through the `otlp/default` exporter. Batching keeps the upstream call rate low while keeping per-event delivery latency under a minute.
4. **Storage and query.** NGINX One Console parses, indexes, and stores the events. The security dashboard and the analytics API both read from the same store.

Every event carries the support ID, the policy that matched it, the violation and signature details, and the request context (method, URL, host, client IP, X-Forwarded-For, geolocation). Those fields are what the dashboard groups and filters on.

---

> ## How data is scoped

Reviewer: ```suggestion
## Data scoping
```

Reviewer 2: done.

Every security event carries enough context to attribute it to a specific data plane, a specific policy, and the application it targeted.

- **Instance**: Each event records the NGINX instance hostname that produced it. Use this to scope by data plane.
- **Policy**: Each event records the F5 WAF for NGINX policy that produced it, so you can compare activity across policy versions or rollouts.

> - **Destination hostname**: Each event records the HTTP `Host` header sent by the client. Use this to scope by the application being attacked, which is often distinct from the instance hostname when the same data plane fronts multiple applications.

Reviewer: ```suggestion
- **Policy**: Each event records the F5 WAF for NGINX policy that produced it. This lets you compare activity across policy versions or rollouts, which is useful when evaluating the effect of a policy change.
- **Destination hostname**: Each event records the HTTP `Host` header sent by the client. This lets you scope by the application being attacked, which is often distinct from the instance hostname when the same data plane fronts multiple applications.
```

explain why it matters

Reviewer 2: policy name - useful for measuring the impact of a policy change before promoting it. Also we have new features lined up in the future to allow customers to link policy seen in the event to managed policy.

Reviewer 2: Destination hostname: this is added to distinguish between hostname name which is infra and the request host header which is app on the host. 

May be a tighter version:

```
Destination hostname: Each event records the HTTP Host header sent by the client that represents the application being attacked, not the instance hostname (which identifies the data plane). Use this when one data plane fronts multiple applications and you need to scope by app rather than by infrastructure.
```

Reviewer 2: updated.

---

> ## Retention

Reviewer: ```suggestion
## ## Event retention

Reviewer 2: taken

Security events are retained for **90 days**. Queries that reach further back than 90 days return no results. If you need long-term retention, forward events to an external Security Information and Event Management (SIEM) system with a [custom log profile]({{< ref "/nginx-one-console/waf-integration/log-profiles/_index.md" >}}) in addition to the `secops_dashboard` log profile.

---

## Use cases

### Operator: triage an active attack

> An operator notices a spike in attack volume on the security dashboard. They use the global filters to narrow down to the affected policy and time window, then drill into the top signatures and attacked endpoints to identify which signatures fired and which URLs were targeted. From a single event, they pull the Support ID, the X-Forwarded-For chain, and the raw request to confirm the source and decide whether to tighten the policy.

Reviewer: ```suggestion
An operator notices a spike in attack volume on the security dashboard. The operator uses the global filters to narrow down to the affected policy and time window, then drills into the top signatures and attacked endpoints to identify which signatures fired and which URLs were targeted. From a single event, the operator pulls the Support ID, the X-Forwarded-For chain, and the raw request to confirm the source and decide whether to tighten the policy.
```

Reviewer 2: considered.

### Security engineer: tune a noisy policy

> A security engineer suspects a policy is producing false positives. They open the security dashboard, filter by policy and blocked requests, and review the breakdown of triggered signatures. The high-volume signatures with low risk and accuracy stand out as candidates for tuning. They cross-check a few events against the raw requests to confirm the signature is firing on legitimate traffic before adjusting the policy.

Reviewer: ```suggestion
A security engineer suspects a policy is producing false positives. The engineer opens the security dashboard, filter by policy and blocked requests, and reviews the breakdown of triggered signatures. The high-volume signatures with low risk and accuracy stand out as candidates for tuning. The engineer cross-checks a few events against the raw requests to confirm the signature is firing on legitimate traffic before adjusting the policy.
```

Reviewer 2: considered.

### Platform team: report on WAF activity across the fleet

> A platform team needs a weekly summary of WAF activity across hundreds of instances. They use the analytics API to pull attack counts, top signatures, and top violations grouped by instance, then render the result in their own reporting tool. The dashboard remains the interactive surface for ad-hoc investigation; the API is the integration point for automation.

Reviewer: ```suggestion
A platform team needs a weekly summary of WAF activity across hundreds of instances. The team uses the analytics API to pull attack counts, top signatures, and top violations grouped by instance, then renders the result in their own reporting tool. The dashboard remains the interactive surface for ad-hoc investigation; the API is the integration point for automation.
```

Reviewer 2: taken

2. **Forwarding.** F5 WAF for NGINX writes the entry over syslog (port `1514` on localhost) using the [`secops_dashboard` log profile]({{< ref "/nginx-one-console/waf-integration/waf-security-dashboard/default-log-profile.md" >}}). NGINX Agent's OpenTelemetry collector receives it through the `tcplog/nginx_app_protect` receiver.
3. **Transport.** The collector batches events and exports them to NGINX One Console through the `otlp/default` exporter. Batching keeps the upstream call rate low while keeping per-event delivery latency under a minute.
4. **Storage and query.** NGINX One Console parses, indexes, and stores the events. The security dashboard and the analytics API both read from the same store.

Every event carries the support ID, the policy that matched it, the violation and signature details, and the request context (method, URL, host, client IP, X-Forwarded-For, geolocation). Those fields are what the dashboard groups and filters on.

---

## How data is scoped

Security events inherit the scope of the instance that produced them.

> - **Tenant**: Events belong to the NGINX One Console tenant the data plane is registered to. Tenants do not see each other's data.

Reviewer 3: seems like can leave out the tenant? customer already operating in their own tenancy, so mentioning here might be exposing internal details?

Reviewer 2: done. Just keeping namespace.

- **Namespace**: Within a tenant, events are partitioned by the namespace the instance lives in. Filters and queries are scoped to the active namespace by default.
- **Instance and policy**: Each event records the instance hostname and the F5 WAF for NGINX policy name that produced it, so you can attribute activity to a specific data plane and a specific policy version.

> - **Hostname vs. request host**: The dashboard exposes both the NGINX instance hostname and the HTTP `Host` header from the request as separate filterable fields. Use the instance hostname to scope by data plane; use the request host to scope by the application being attacked.

Reviewer 4: We have referred to `request_host` in the UI as `Destination Hostname` in filters since that is how it is found in the events detail. Might be good to use that name here

Reviewer 2: "### Add Filter" section covers filters and notes on them. 

| **Destination Hostname** | The HTTP `Host` header sent by the client. Use this to scope by the application being attacked. |

I think it captures host header information correctly.

---

## Retention

> Security events are retained for **90 days**. Queries that reach further back than 90 days return no results. If you need long-term retention, forward events to an external SIEM with a [custom log profile]({{< ref "/nginx-one-console/waf-integration/log-profiles/_index.md" >}}) in addition to the `secops_dashboard` log profile.

Reviewer 3: what's `SIEM` ? i think need to define this somewhere for the first time?

Reviewer 2: expanded and clarified.
